java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java

```diff
@@ -74,7 +74,7 @@
   private final long allocatorManagerId = MANAGER_ID_GENERATOR.incrementAndGet();
   private final int size;
   private final UnsafeDirectLittleEndian underlying;
-  private final IdentityHashMap<BufferAllocator, BufferLedger> map = new IdentityHashMap<>();
+  private final LowCostIdentityHasMap<BufferLedger> map = new LowCostIdentityHasMap<>();
```

Comment:
Let's add comment here on why we use this structure.

Reply:
done